src/search/options.h

```diff
@@ -0,0 +1,82 @@
+#ifndef SEARCH_OPTIONS_H
+#define SEARCH_OPTIONS_H
+
+#include <iostream>
+
+#include <boost/program_options.hpp>
+
+namespace po = boost::program_options;
+
+class Options {
+    std::string filename;
+    std::string generator;
+    std::string search_engine;
+    std::string evaluator;
+    std::string state_representation;
+    int seed;
+
+public:
+    Options(int argc, char** argv) {
+        po::options_description description("Allowed options");
+        description.add_options()
+            ("filename,f", po::value<std::string>()->default_value("output.lifted"), "Lifted task file name.")
+            ("help,h", "Display this help message.")
+            ("seed", po::value<int>()->default_value(1), "Random seed.")
+            ("evaluator,e", po::value<std::string>()->required(), "Heuristic evaluator.")
+            ("generator,g", po::value<std::string>()->required(), "Successor generator method.")
+            ("search,s", po::value<std::string>()->required(), "Search engine.")
+            ("state-representation,r", po::value<std::string>()->default_value("sparse"), "State representation.")
+            ;
+
+        po::variables_map vm;
+
+        try {
+            po::store(po::command_line_parser(argc, argv).options(description).run(), vm);
+
+            if (vm.count("help")) {
+                std::cout << description << "\n";
+                exit(0);
+            }
+            po::notify(vm);
+        } catch(const std::exception& ex) {
+            std::cout << "Error with command-line options:" << ex.what() << std::endl;
+            std::cout << std::endl << description << std::endl;
+            exit(0);
```

Comment:
This should be != 0

Reply:
I copied from your planner :) I will change it.

Reply:
Sounds like copyright infringement!